include/picongpu/fields/laserProfiles/laserExpRampWithPrepulse.hpp

```diff
@@ -0,0 +1,188 @@
+/* Copyright 2013-2017 Axel Huebl, Heiko Burau, Rene Widera, Richard Pausch, Stefan Tietze
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+
+
+namespace picongpu
+{
+/** Wavepaket with spacial gaussian envelope and complicated longitudinal shape.
+ * Allows defining a prepulse and two regions of exponential preramp with inde-
+ * pendent slopes. The definition works by specifying three (t, intensity)-
+ * -points, where time is counted from the very beginning in SI and the inten-
+ * sity (yes, intensity, not amplitude) is given in multiples of the main peak.
+ *
+ * care! Problematic for few cycle laser pulses. Thought the rest is cloned 
+ * from laserWavepacket, the correctionFactor is not included (this was a 
```

Comment:
A user knows nothing about the implementation of "laserWavepacket", only the interface. Therefor, a user does not know any "correctionFactor". (Side note: I think you also got a bit wrong what it did there, it's described good in `laserPlaneWave`: it ensures the integral is well-defined which has e.g. an influence on a single test particle's final momentum/position when being pushed through the full laser.) Instead, describe in what parameter range you can use the laser pulse and in what not (e.g. few-cycle) and why (not well-defined cut-off).

Reply:
but neither is this comment exposed to the user. Shall I completely remove it, and just add the appropriate comments in the .param

Reply:
the comment about the correction factor should *not* go into the `.param` at all, since this is not exposed to the user.

here you can make a reference to the wavepacket if really needed.